### DISCUSSION TITLE
Alert report contains VNode

### DISCUSSION THREAD

**Discussion by jokerZhou** [apache/skywalking#10757] [General]
I use alert report logger request,but the endpoint always contains a vode,
I find VNode in FAQ : VNode is created by the UI itself, rather than being reported by the agent or tracing SDK. It indicates that some spans are missed in the trace data in this query.
so Endpoint contains vnode is incorrect. but in metrics endpoint_relation_resp_time will report alarm that have vnode.I  want use alarm report longger request . but always report alarm with VNode,

**Comment by wu-sheng** [comment]
I am afraid this is impossible. VNode is a concept that showing in the query only.
The reference of span could be reported in different OAPs with uncertain order. 
There never be perfect time to determine this.

**Reply by jokerZhou** [reply]
Thanks for you  reply .
So I can't get correct alarm of long request.
How to get alarm of long request from skywalking?

**Reply by jokerZhou** [reply]
How skip include VNode trace?

**Reply by wu-sheng** [reply]
I don't know what you mean. There is no VNode trace. VNode is actually not existing in the backend.

**Reply by jokerZhou** [reply]
Sorry , My English is not good What I mean is that I use Skywalking Alert warnings on my system. I want to monitor interfaces that exceed a long time. However, push warnings are usually caused by VNodes. From the UI, I find the trace of VNode usually lasts for a long time, does this trigger a warning?

**Reply by jokerZhou** [reply]
额 我英语不好, 就我发现触发警告.我使用的alert 规则是:
rules:
  endpoint_relation_resp_time_rule:
    metrics-name: endpoint_relation_resp_time
    threshold: 10000
    op: ">"
    period: 10
    count: 1
    message: 端点 {name} 响应时间超过10秒钟
但是出现警告我再UI上看 发现是trace中包含VNode 而且每次有VNode的trace通常持续时间会较长
![image](https://user-images.githubusercontent.com/4216871/236358659-2288d825-b3de-4ff9-9b34-cc554af11ced.png)

**Reply by wu-sheng** [reply]
No, VNodes never created alerting, no matter what you set.

**Reply by jokerZhou** [reply]
Thank you .  I will  find my wrong.
The other question. 
The other question.How to get trace Id by alert?
I want to get the trace that trigger the alert in skywalking ui. becouse so many long time trace that cuase by vnode in UI . I  difficult to find the trace with the alert.

**Reply by wu-sheng** [reply]
I think I replied many times. Alerting is triggered by the metrics, rather than a single request or trace.
That concept is totally wrong in a distributed product env.

**Reply by jokerZhou** [reply]
Thank you . I'm understand..
